patches/minecraft/net/minecraft/entity/monster/EndermanEntity.java.patch

```diff
@@ -16,6 +16,15 @@
     }
  
     protected void func_70088_a() {
+@@ -159,7 +159,7 @@
+ 
+    private boolean func_70821_d(PlayerEntity p_70821_1_) {
+       ItemStack itemstack = p_70821_1_.field_71071_by.field_70460_b.get(3);
+-      if (itemstack.func_77973_b() == Blocks.field_196625_cS.func_199767_j()) {
++      if (itemstack.isEnderMask()) {
```

Comment:
Since we have it, it is probably worth passing in both the enderman entity and player entity to allow more power.

Reply:
I don't really understand what you mean by that?

Reply:
In the function parameters, he wants to pass the Enderman entity as well as the Player entity. That's how I read it.

Reply:
So basically do a generic method shouldAttackPlayer?

Reply:
Hmm. I guess my guess must have been wrong, because I assumed he wanted
 `shouldAttackPlayer(PlayerEntity player, EndermanEntity enderman)`

But then you'd have something like 
`if (entity instanceof EndermanEntity) {
    (EndermanEntity)entity.shouldAttackPlayer(player, (EndermanEntity)entity)`

Passing an object to a function of itself is usually bad practice, so i'm not sure what he meant.

Reply:
Yes it's not wrong @TheCurle

Reply:
So @LexManos , I can't understand what you're suggesting

Reply:
Lex is suggesting that you make `isEnderMask` be like `isEnderMask(EndermanEntity enderman, PlayerEntity player)`

Reply:
Okay, but what's the use of doing it like this? Because I do not see the possible use in this form.

Reply:
Lex's point is that given we have the information anyways we may as well pass it. That way if someone does have use case like they only want specific players to have it work as an ender mask, or only specific endermen (or subspecies of enderman that get added by some mod)

Reply:
So basically put just 2 arguments in the function, ok I understand better.